MekHQ/src/mekhq/campaign/universe/factionStanding/FactionStandingUltimatum.java

```diff
@@ -0,0 +1,195 @@
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * A copy of the GPL should have been included with this project;
+ * if not, see <https://www.gnu.org/licenses/>.
+ *
+ * NOTICE: The MegaMek organization is a non-profit group of volunteers
+ * creating free software for the BattleTech community.
+ *
+ * MechWarrior, BattleMech, `Mech and AeroTech are registered trademarks
+ * of The Topps Company, Inc. All Rights Reserved.
+ *
+ * Catalyst Game Labs and the Catalyst Game Labs logo are trademarks of
+ * InMediaRes Productions, LLC.
+ *
+ * MechWarrior Copyright Microsoft Corporation. MekHQ was created under
+ * Microsoft's "Game Content Usage Rules"
+ * <https://www.xbox.com/en-US/developers/rules> and it is not endorsed by or
+ * affiliated with Microsoft.
+ */
+package mekhq.campaign.universe.factionStanding;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.Map;
+
+import megamek.common.enums.Gender;
+import mekhq.campaign.Campaign;
+import mekhq.campaign.personnel.Person;
+import mekhq.campaign.personnel.enums.PersonnelRole;
+import mekhq.gui.dialog.factionStanding.FactionStandingUltimatumDialog;
+
+/**
+ * Handles the orchestration and presentation of faction standing ultimatum events.
+ *
+ * <p>This class manages the association between significant historical dates and their respective Faction Standing
+ * ultimatum scenarios, triggering appropriate in-game dialogs and transitions when such events are detected for the
+ * campaign's current faction. It ensures scenarios such as major faction splits or leadership transitions (e.g., the
+ * Federated Commonwealth Civil War, ComStar Schism) are detected and surfaced to the player, accompanied by unique
+ * dialog sequences involving prominent historical personalities.</p>
+ *
+ * <p>Key responsibilities include:</p>
+ * <ul>
+ *     <li>Tracking historically significant faction ultimatum dates and their context.</li>
+ *     <li>Providing static checks for whether an ultimatum event is relevant to a given date and faction.</li>
+ *     <li>Validating campaign and faction alignment before presenting a scenario dialog to the player.</li>
+ *     <li>Constructing {@link Person} representations of event participants for use in dialogs.</li>
+ * </ul>
+ *
+ * <p>See also: {@link FactionStandingUltimatumData} and {@link FactionStandingAgitatorData} for structure of scenario
+ * and participant data.</p>
+ *
+ * @author Illiani
+ * @since 0.50.07
+ */
+public class FactionStandingUltimatum {
+    /**
+     * Represents the earliest possible date for when Lord Espinosa overthrew the legitimate heir of the Arano
+     * Coalition
+     *
+     * <p>This date was determined by comparing dates in the House Arano manual, as well as dates in HBS'
+     * Battletech.</p>
+     */
+    private static final LocalDate ESPINOSA_COUP = LocalDate.of(3022, Month.FEBRUARY, 1);
```

Comment:
If we add any more of these events, I think we should discuss making a loadable class and a yaml file format for these configurations.